### ISSUE TITLE
Cannot access offset of type string on string

### ISSUE THREAD

**Issue by jesse-deboer** [magmodules/magento2-reloadify#15]
I am getting the issue that was reported before: https://github.com/magmodules/magento2-reloadify/issues/13

These errors not only occur in getProfiles but also the following:

**getCategories()**

Route example: `rest/V1/reloadify/language/4/categories`
```
TypeError: Cannot access offset of type string on string in /data/web/releases/20230111083735/vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php:373
Stack trace:
#0 /data/web/releases/20230111083735/vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php(405): Magento\Eav\Model\Entity\Collection\AbstractCollection->addAttributeToFilter()
#1 /data/web/releases/20230111083735/vendor/magento/framework/Api/SearchCriteria/CollectionProcessor/FilterProcessor.php(89): Magento\Eav\Model\Entity\Collection\AbstractCollection->addFieldToFilter()
#2 /data/web/releases/20230111083735/vendor/magento/framework/Api/SearchCriteria/CollectionProcessor/FilterProcessor.php(48): Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor->addFilterGroupToCollection()
#3 /data/web/releases/20230111083735/vendor/magento/framework/Api/SearchCriteria/CollectionProcessor.php(38): Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor->process()
#4 /data/web/releases/20230111083735/vendor/magmodules/magento2-reloadify/Service/WebApi/Category.php(104): Magento\Framework\Api\SearchCriteria\CollectionProcessor->process()
#5 /data/web/releases/20230111083735/vendor/magmodules/magento2-reloadify/Service/WebApi/Category.php(64): Magmodules\Reloadify\Service\WebApi\Category->getCollection()
#6 /data/web/releases/20230111083735/vendor/magmodules/magento2-reloadify/Model/WebApi/Repository.php(234): Magmodules\Reloadify\Service\WebApi\Category->execute()
#7 [internal function]: Magmodules\Reloadify\Model\WebApi\Repository->getCategories()
#8 /data/web/releases/20230111083735/vendor/magento/module-webapi/Controller/Rest/SynchronousRequestProcessor.php(95): call_user_func_array()
#9 /data/web/releases/20230111083735/vendor/magento/module-webapi/Controller/Rest.php(195): Magento\Webapi\Controller\Rest\SynchronousRequestProcessor->process()
#10 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Webapi\Controller\Rest->dispatch()
#11 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Webapi\Controller\Rest\Interceptor->___callParent()
#12 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Webapi\Controller\Rest\Interceptor->Magento\Framework\Interception\{closure}()
#13 /data/web/releases/20230111083735/generated/code/Magento/Webapi/Controller/Rest/Interceptor.php(23): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins()
#14 /data/web/releases/20230111083735/vendor/magento/framework/App/Http.php(116): Magento\Webapi\Controller\Rest\Interceptor->dispatch()
#15 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Http->launch()
#16 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\Http\Interceptor->___callParent()
#17 /data/web/releases/20230111083735/vendor/justbetter/magento2-sentry/Plugin/GlobalExceptionCatcher.php(78): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#18 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(135): JustBetter\Sentry\Plugin\GlobalExceptionCatcher->aroundLaunch()
#19 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#20 /data/web/releases/20230111083735/generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http\Interceptor->___callPlugins()
#21 /data/web/releases/20230111083735/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#22 /data/web/releases/20230111083735/pub/index.php(30): Magento\Framework\App\Bootstrap->run()
#23 {main}
```

**getVariants()**

Route example: `rest/V1/reloadify/language/4/variants`
```
TypeError: Cannot access offset of type string on string in /data/web/releases/20230111083735/vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php:373
Stack trace:
#0 /data/web/releases/20230111083735/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php(1596): Magento\Eav\Model\Entity\Collection\AbstractCollection->addAttributeToFilter()
#1 /data/web/releases/20230111083735/vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php(405): Magento\Catalog\Model\ResourceModel\Product\Collection->addAttributeToFilter()
#2 /data/web/releases/20230111083735/vendor/magento/framework/Api/SearchCriteria/CollectionProcessor/FilterProcessor.php(89): Magento\Eav\Model\Entity\Collection\AbstractCollection->addFieldToFilter()
#3 /data/web/releases/20230111083735/vendor/magento/framework/Api/SearchCriteria/CollectionProcessor/FilterProcessor.php(48): Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor->addFilterGroupToCollection()
#4 /data/web/releases/20230111083735/vendor/magento/framework/Api/SearchCriteria/CollectionProcessor.php(38): Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor->process()
#5 /data/web/releases/20230111083735/vendor/magmodules/magento2-reloadify/Service/WebApi/Variants.php(193): Magento\Framework\Api\SearchCriteria\CollectionProcessor->process()
#6 /data/web/releases/20230111083735/vendor/magmodules/magento2-reloadify/Service/WebApi/Variants.php(107): Magmodules\Reloadify\Service\WebApi\Variants->getCollection()
#7 /data/web/releases/20230111083735/vendor/magmodules/magento2-reloadify/Model/WebApi/Repository.php(192): Magmodules\Reloadify\Service\WebApi\Variants->execute()
#8 [internal function]: Magmodules\Reloadify\Model\WebApi\Repository->getVariants()
#9 /data/web/releases/20230111083735/vendor/magento/module-webapi/Controller/Rest/SynchronousRequestProcessor.php(95): call_user_func_array()
#10 /data/web/releases/20230111083735/vendor/magento/module-webapi/Controller/Rest.php(195): Magento\Webapi\Controller\Rest\SynchronousRequestProcessor->process()
#11 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Webapi\Controller\Rest->dispatch()
#12 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Webapi\Controller\Rest\Interceptor->___callParent()
#13 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Webapi\Controller\Rest\Interceptor->Magento\Framework\Interception\{closure}()
#14 /data/web/releases/20230111083735/generated/code/Magento/Webapi/Controller/Rest/Interceptor.php(23): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins()
#15 /data/web/releases/20230111083735/vendor/magento/framework/App/Http.php(116): Magento\Webapi\Controller\Rest\Interceptor->dispatch()
#16 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Http->launch()
#17 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\Http\Interceptor->___callParent()
#18 /data/web/releases/20230111083735/vendor/justbetter/magento2-sentry/Plugin/GlobalExceptionCatcher.php(78): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#19 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(135): JustBetter\Sentry\Plugin\GlobalExceptionCatcher->aroundLaunch()
#20 /data/web/releases/20230111083735/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#21 /data/web/releases/20230111083735/generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http\Interceptor->___callPlugins()
#22 /data/web/releases/20230111083735/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#23 /data/web/releases/20230111083735/pub/index.php(30): Magento\Framework\App\Bootstrap->run()
#24 {main}
```

**Comment by Marvin-Magmodules** [comment]
Hi @jesse-deboer,

We are happy to share that we’ve just released the new [1.5.5](https://github.com/magmodules/magento2-reloadify/releases/tag/1.5.5) version where we have fixed this issue. Thank you for opening this issue and for the detailed issue report. 

We are closing this issue now but please feel free to reopen the issue if this still occurs.

**Comment by peterjaap** [comment]
@Marvin-Magmodules thanks, 1.5.5 indeed fixed it;

![image](https://user-images.githubusercontent.com/431360/214293358-1a31a660-a00f-430b-9024-760af683d7d8.png)
